subprojects/core/src/main/java/org/gradle/api/internal/file/AbstractFileCollection.java

```diff
@@ -193,6 +194,12 @@ public Object asType(Class<?> type) throws UnsupportedOperationException {
         return DefaultGroovyMethods.asType(this, type);
     }
 
+    @Override
+    public TaskDependency getBuildDependencies() {
+        DeprecationLogger.nagUserOfDiscontinuedMethod("AbstractFileCollection.getBuildDependencies()");
```

Comment:
I feels somewhat strange to deprecate a method on an internal class.

If we really want to add a deprecation message, we should at least say what we expect plugin authors to do, i.e. implement the method and don't use the method in here.

Reply:
I really would like to remove this method in 5.0, so showing _some_ deprecation message makes sense. However, I don't want to recommend that they implement the method, because, well, they shouldn't implement `FileCollection` on their own anyway. Suggestions?

Reply:
Then the suggestion of the deprecation method would be: "Do not implement FileCollection", WDYT?